lib/withOnyx.js

```diff
@@ -37,15 +37,28 @@ export default function (mapOnyxToState) {
                 // disconnected. It is a key value store with the format {[mapping.key]: connectionID}.
                 this.activeConnectionIDs = {};
 
+                const cachedState = {};
+
+                // Check if we can get all the data synchronously from cache
+                _.each(mapOnyxToState, (mapping, propertyName) => {
+                    const key = Str.result(mapping.key, props);
+                    const value = Onyx.tryGetCachedValue(key, mapping);
+
+                    if (value !== undefined) {
+                        cachedState[propertyName] = value;
+                    }
+                });
+
+                // If we have all the data we need, then we can render the component immediately
+                cachedState.loading = _.keys(cachedState).length < requiredKeysForInit.length;
+
                 // Object holding the temporary initial state for the component while we load the various Onyx keys
-                this.tempState = {};
+                this.tempState = cachedState;
 
-                this.state = {
-                    // If there are no required keys for init then we can render the wrapped component immediately
-                    loading: requiredKeysForInit.length > 0,
-                };
+                this.state = cachedState;
             }
 
+            // TODO: I think we don't have to wait until mount to subscribe to Onyx keys. We can do it in the constructor as well.
```

Comment:
Random (impossible to know) Expensify rule incoming, but we never put "TODO" in the code anywhere. something something it is better to make an issue so that we actually do whatever the "TODO" asks us to do.

Reply:
FWIW, I can't remember but this might _possibly_ be how we had it at some point. The data can be returned very fast and the component could takes a while to mount - then we end up setting state on an unmounted component (which throws the ugly warning). Honestly though - can't remember.

Reply:
yeah, I think this change would be a separate PR as well where we can look at and discuss all those cases (however, I might propose a bit of a architectural change to onyx in the following days which would handle this point as well)